### ISSUE TITLE
CloudFormation.Stackset - Plan Requires AdministrationRoleARN while Apply says "not supported"

### ISSUE THREAD

**Issue by crispyblock** [hashicorp/terraform-provider-aws#12125]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version 12.21 
```
Downloading plugin for provider "aws" (hashicorp/aws) 2.49.0...
```

### Affected Resource(s)

* aws_cloudformation_stack_set
Specifically, the administration_role_arn is acting inconsistently

### Terraform Configuration Files

```hcl
resource "aws_cloudformation_stack_set" "tgrc_platform_owner_account_access" {
  administration_role_arn = data.aws_iam_role.AWSCloudFormationStackSetsOrgAdminRole.arn
  name                    = "platform-owner-account-access"
  description             = "AWS CloudFormation IAM Role for Platform Owner Pivot access. It creates a single role granting assume access to the TrustedEntityArn parameter. "
  depends_on              = [aws_organizations_organization.org, ]
  #   permissions_model = SERVICE_MANAGED
  execution_role_name = "stacksets-exec-423b171060aaf4602fdc308b77d46dd8"
  parameters = {
    AuthorizedAssumerRoleArn = var.RoleArn
  }
  capabilities = [
    "CAPABILITY_NAMED_IAM",
  ]
  tags = merge(
    var.common_tags,
    {
      Name           = "Account Bootstrap Automation",
      AppName        = "Platform-Bootstrap",
      AppEnvironment = local.environment
    },
  )
  template_body = data.local_file.account-access-cloudformation-body.content
}
```

### Expected Behavior

Apply Succeeds

### Actual Behavior

*Case Having Attribute* - Validation Error during Apply

*Case Without Attribute* - Error: Missing required argument. During Plan

#### Case Having Attribute

1. `terraform plan` - Indicates no changes are being made to these attributes...
``` ~ resource "aws_cloudformation_stack_set" "tgrc_platform_owner_account_access" {
442 | administration_role_arn = "arn:aws:iam::913701052451:role/aws-service-role/stacksets.cloudformation.amazonaws.com/AWSServiceRoleForCloudFormationStackSetsOrgAdmin"
443 | arn                     = "arn:aws:cloudformation:us-east-1:xxxxxxxxxxx:stackset/tgrc-platform-owner-account-access:b9b11041-6539-4676-afca-89a46c3bca88"
444 | capabilities            = [
445 | "CAPABILITY_NAMED_IAM",
446 | ]
447 | description             = "AWS CloudFormation IAM Role for Platform Owner Pivot access."
448 | execution_role_name     = "stacksets-exec-423b171060aaf4602fdc308b77d46dd8"
449 | id                      = "tgrc-platform-owner-account-access"
450 | name                    = "tgrc-platform-owner-account-access"
```

2. `terraform apply`  - Throws this validation error
```
Error: error updating CloudFormation Stack Set (tgrc-platform-owner-account-access): ValidationError: AdministrationRoleARN not supported
--
143 | status code: 400
```

#### Case Without Attribute

1. This time I comment out the attribute from the resource (coded above)
```hcl
resource "aws_cloudformation_stack_set" "tgrc_platform_owner_account_access" {
  # administration_role_arn = data.aws_iam_role.AWSCloudFormationStackSetsOrgAdminRole.arn
  name                    = "platform-owner-account-access"
blah, blah
```
2. During terraform Plan I receive this error.
```
Error: Missing required argument
101 |  
102 | on cloudformation_stackset.tf line 45, in resource "aws_cloudformation_stack_set" "tgrc_platform_owner_account_access":
103 | 45: resource "aws_cloudformation_stack_set" "tgrc_platform_owner_account_access" {
104 |  
105 | The argument "administration_role_arn" is required, but no definition was
106 | found.

```
### Important Factoids
This is being applied to an organizational master account. 
thus the permissions_model = SERVICE_MANAGED is set up.
### References

* #0000

**Comment by jonshern** [comment]
Looks like it is required.  But may need to set to optional.

--administration-role-arn (string)

The Amazon Resource Number (ARN) of the IAM role to use to update this stack set.
Specify an IAM role only if you are using customized administrator roles to control which users or groups can manage specific stack sets within the same administrator account. For more information, see Granting Permissions for Stack Set Operations in the AWS CloudFormation User Guide .
If you specified a customized administrator role when you created the stack set, you must specify a customized administrator role, even if it is the same customized administrator role used with this stack set previously.

The terraform indicates it is required. 

`			"administration_role_arn": {
				Type:         schema.TypeString,
				Required:     true,
				ValidateFunc: validateArn,
`
https://github.com/terraform-providers/terraform-provider-aws/blob/ef098db34234c4f533be2a3f5063bf609e960337/aws/resource_aws_cloudformation_stack_set.go#L32

**Comment by marcelboettcher** [comment]
@jonshern It looks like they changed the cli documentation and it is not required anymore with the integration into aws organizations.

**cli docs**:

--administration-role-arn (string)

The Amazon Resource Number (ARN) of the IAM role to use to create this stack set.

Specify an IAM role only if you are using customized administrator roles to control which users or groups can manage specific stack sets within the same administrator account. For more information, see Prerequisites: Granting Permissions for Stack Set Operations in the AWS CloudFormation User Guide .


I have the issue that terraform requires the property, but I am not allowed to pass the role from my deployment role (admin access). The AWSServiceRoleForCloudFormationStackSetsOrgAdmin just trusts cloudformation service, so I guess the administration-role-arn must be optional!

Error: error creating CloudFormation Stack Set: AccessDenied: User: arn:aws:sts::XXX:assumed-role/AWSReservedSSO_ROLENAME/USERX is not authoriz
ed to perform: iam:PassRole on resource: arn:aws:iam::XXX:role/aws-service-role/stacksets.cloudformation.amazonaws.com/AWSServiceRoleForCloudFormationStackSetsOrgAdmin
        status code: 403

**Comment by bdwyertech** [comment]
https://github.com/aws/aws-sdk-go/commit/de8dd8e6d2f89bf6a0363a2c3cc8ad76381bf5be

**Comment by bdwyertech** [comment]
Related: #12422  #12425

**Comment by bflad** [comment]
Support for `SERVICE_MANAGED` CloudFormation Stack Sets has been merged as part of #12422 and will release with version 3.35.0 of the Terraform AWS Provider later this week. 👍
